ion-schema/src/isl/isl_range.rs

```diff
@@ -525,8 +525,18 @@ impl<T: std::cmp::PartialOrd> RangeImpl<T> {
         end: TypedRangeBoundaryValue,
     ) -> IonSchemaResult<RangeImpl<usize>> {
         match (start, end) {
-            (TypedRangeBoundaryValue::NonNegativeInteger(v1), TypedRangeBoundaryValue::NonNegativeInteger(v2)) => {
-                RangeImpl::range(v1, v2)
+            (TypedRangeBoundaryValue::NonNegativeInteger(Value(v1, v1_type)), TypedRangeBoundaryValue::NonNegativeInteger(Value(v2, v2_type))) => {
+                // verify this is not an empty range for which there is no valid non negative integer values
+                if v2 > v1 && v2 - v1 == 1
+                    && v1_type == RangeBoundaryType::Exclusive
+                    && v2_type == RangeBoundaryType::Exclusive
```

Comment:
I was thinking to do the same but since RangeImpl is generic, I would have to unwrap `RangeImpl<T>` to get a particular range type and then perform `is_empty()` check based on `T`. Which is why adding it here made sense. But that being said I could add something similar in `TypedRangeBoundary`.